### PR TITLE
Navigation Dropdown feedback Changes

### LINKED ISSUE #6257: [Task] NavigationBar Menu Feedback changes
This is an issue to track Navigation bar menu changes requested for feedback,
Total feedback changes:

- Align Items Vertically

- Left and right padding for text and icon were not symmetrical

- Style of delete text should match delete menu in Applications page

- Remove Shortcuts

- Rotate Chevron icon on opening and closing state

### PR DESCRIPTION
**Pull request by rahulramesha**
## Description

This PR addresses some Feedback on NavigationMenu dropdown

Fixes #6257

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

UI verification

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: Task/5463_Navigation_dropdown 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 51.69 **(-0.02)** | 33.63 **(-0.01)** | 29.53 **(-0.03)** | 52.35 **(-0.01)**
 :green_circle: | app/client/src/pages/Editor/ToggleModeButton.tsx | 64.54 **(0.25)** | 26.15 **(0)** | 40 **(0)** | 62.79 **(0)**
 :red_circle: | app/client/src/pages/Editor/EditorAppName/NavigationMenuData.ts | 62.86 **(-1.24)** | 12.5 **(0)** | 10 **(0.91)** | 61.76 **(-1.4)**
 :green_circle: | app/client/src/pages/Editor/EditorAppName/NavigationMenuItem.tsx | 24.07 **(1.15)** | 24 **(2.95)** | 6.67 **(-2.42)** | 26 **(1)**
 :red_circle: | app/client/src/pages/Editor/EditorAppName/index.tsx | 51.35 **(-1.43)** | 26.92 **(-2.25)** | 3.7 **(-0.3)** | 46.15 **(-1.47)**
 :red_circle: | app/client/src/utils/helpers.tsx | 53.05 **(-1.83)** | 26.03 **(-4.11)** | 21.43 **(-3.57)** | 46.51 **(-1.55)**
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 50.71 **(0.24)** | 38.6 **(0.88)** | 36.21 **(0)** | 55.35 **(0.27)**</details>

### CODE REVIEW

File: app/client/src/pages/Editor/EditorAppName/index.tsx [hunk 1]
```diff
@@ -75,8 +75,8 @@ const Container = styled.div<{ isPopoverOpen: boolean }>`
   &&&& .${Classes.EDITABLE_TEXT_CONTENT}, &&&& .${Classes.EDITABLE_TEXT_INPUT} {
     display: block;
     ${(props) => getTypographyByKey(props, "h4")};
-    line-height: 19px !important;
-    padding: 8px 5px;
+    line-height: ${(props) => props.theme.smallHeaderHeight} !important;
+    padding: 0 10px;
```
Comment (rishabhsaxena) [inline app/client/src/pages/Editor/EditorAppName/index.tsx:79]: Nipick: we can use `theme.spaces` here

**Review by rishabhsaxena — DISMISSED** [review]
One minor change: we should hide the option to change the view mode from pages other than the editor since we don't have comment options for other pages yet

**Comment by rahulramesha** [comment]
@rishabhsaxena for view modes, the dropdown is only shown in Editor mode and even in Editor mode the view modes are shown only when Comments are enabled. i.e., in the same scenario where Toggle Mode Buttons are shown.

**Comment by rishabhsaxena** [comment]
> @rishabhsaxena for view modes, the dropdown is only shown in Editor mode and even in Editor mode the view modes are shown only when Comments are enabled. i.e., in the same scenario where Toggle Mode Buttons are shown.

Sorry I meant it should show only on the widgets editor page and we should hide it for apis, queries, datasources. In the `isCommentMode` we have this logic, we could move it to a util and use it at both of those functions

**Review by rishabhsaxena — APPROVED** [review]

File: app/client/src/pages/Editor/EditorAppName/index.tsx (hunk 1 above)
Comment (rahulramesha) [reply app/client/src/pages/Editor/EditorAppName/index.tsx]: Changed

**Review by rishabhsaxena — APPROVED** [review]

File: app/client/src/pages/Editor/EditorAppName/NavigationMenuData.ts [hunk 2]
```diff
@@ -26,10 +27,9 @@ export const GetNavigationMenuData = ({
   currentDeployLink,
   deploy,
   editMode,
-  theme,
 }: NavigationMenuDataProps): MenuItemData[] => {
   const dispatch = useDispatch();
-  const commentsEnabled = useSelector(areCommentsEnabledForUserAndApp);
+  const isShouldHideComments = useShouldHideComments();
```
Comment (SatishGandham) [inline app/client/src/pages/Editor/EditorAppName/NavigationMenuData.ts:32]: The name looks weird

File: app/client/src/pages/Editor/ToggleModeButton.tsx [hunk 3]
```diff
@@ -320,16 +320,17 @@ const useShowCommentDiscoveryTooltip = (): [boolean, typeof noop] => {
   ];
 };
 
-const useShouldHide = () => {
+export const useShouldHideComments = () => {
```
Comment (SatishGandham) [inline app/client/src/pages/Editor/ToggleModeButton.tsx:323]: Are we using names like this anywhere else? I haven't seen should in variable names before.
```
areCommentsHidden
showComments

```

**Review by SatishGandham — APPROVED** [review]

**Comment by vivekverma2312** [comment]
@rahulramesha LGTM on deploy preview.

**Review by SatishGandham — APPROVED** [review]

File: app/client/src/pages/Editor/ToggleModeButton.tsx (hunk 3 above)
Comment (rahulramesha) [reply app/client/src/pages/Editor/ToggleModeButton.tsx]: Agreed

File: app/client/src/pages/Editor/EditorAppName/NavigationMenuData.ts (hunk 2 above)
Comment (rahulramesha) [reply app/client/src/pages/Editor/EditorAppName/NavigationMenuData.ts]: Agreed

**Review by SatishGandham — APPROVED** [review]

**Pull request merged**